### PR TITLE
Modify rule S5754: LaYC format

### PR DESCRIPTION
**Pull request by joke1196**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S5754/python/rule.adoc [hunk 1]
```diff
@@ -1,74 +1,76 @@
+This rule raises an issue when a bare ``++except:++``, an ``++except BaseException++`` or an ``++except SystemExit++`` block don't re-raise the exception caught.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S5754/python/rule.adoc:1]: Shouldn't it be singular (doesn't) instead of don't?

File: rules/S5754/python/rule.adoc [hunk 2]
```diff
@@ -1,74 +1,76 @@
...
+
 == Why is this an issue?
 
-https://docs.python.org/3/library/exceptions.html#SystemExit[``++SystemExit++``] is raised when https://docs.python.org/3/library/sys.html#sys.exit[``++sys.exit()++``] is called. This exception is expected to propagate up until the application stops. It is ok to catch it when a clean-up is necessary but it should be raised again immediately.
+A https://docs.python.org/3/library/exceptions.html#SystemExit[``++SystemExit++``] exception is raised when https://docs.python.org/3/library/sys.html#sys.exit[``++sys.exit()++``] is called. 
+This exception is used to signal the interpreter to exit. The exception is expected to propagate up until the program stops. 
+It is possible to catch this exception in order perform, for example, clean-up tasks but it should be raised again to allow the interpreter to exit as expected. Not re-raising such exception could lead to undesired behaviour.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S5754/python/rule.adoc:7]: missing "to"; in order to perform.
Also maybe we could remove the "but" and make it a separate sentence for readability: It should, however, be raised again to allow the interpreter to exit as expected.

File: rules/S5754/python/rule.adoc [hunk 3]
```diff
@@ -1,74 +1,76 @@
...
 
 
-A https://docs.python.org/3/reference/compound_stmts.html#the-try-statement[bare ``++except:++`` statement], i.e. an ``++except++`` without any exception class, is equivalent to https://docs.python.org/3/library/exceptions.html#BaseException[``++except BaseException++``]. Both statements will catch every exception, including ``++SystemExit++``. It is recommended to catch instead a specific exception. If it is not possible, the exception should be raised again.
+A https://docs.python.org/3/reference/compound_stmts.html#the-try-statement[bare ``++except:++`` statement], i.e. an `except` block without any exception class, is equivalent to https://docs.python.org/3/library/exceptions.html#BaseException[``++except BaseException++``]. 
+Both statements will catch every exceptions, including `SystemExit`. It is recommended to catch instead a more specific exception, or if it is not possible, the exception should be raised again.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S5754/python/rule.adoc:11]: Here also I'd suggest splitting the sentences instead of using "or".

File: rules/S5754/python/rule.adoc [hunk 4]
```diff
@@ -1,74 +1,76 @@
...
 
 
-Note that it is also a good idea to reraise the https://docs.python.org/3/library/exceptions.html#KeyboardInterrupt[``++KeyboardInterrupt++``] exception.
+It is also a good idea to re-raise the https://docs.python.org/3/library/exceptions.html#KeyboardInterrupt[``++KeyboardInterrupt++``] exception. Similarly to `SystemExit`,`KeyboardInterrupt` is used to signal the interpreter to exit. Not re-raising such exception could also lead to undesired behaviour.  
 
+== How to fix it
 
-This rule raises an issue when a bare ``++except:++``, an ``++except BaseException++`` or an ``++except SystemExit++`` don't reraise the exception caught.
+Re-raise `SystemExit` exceptions, `BaseException` exceptions and exceptions caught in a bare `except` statement.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S5754/python/rule.adoc:18]: I'd suggest not repeating `exceptions` and reword to:
Re-raise `SystemExit`, `BaseException` and any exception caught in a bare `except` clause.

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
Looks good!
Minor typos/rewording suggestions.

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM

**Review by jean-jimbo-sonarsource — APPROVED** [review]
LGTM

**Review by maksim-grebeniuk-sonarsource — APPROVED** [review]

**Pull request merged**